The job **`i686-gnu-nopt-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/27336754404/job/80763048485) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/80763048485) [(plain)](https://github.com/rust-lang/rust/commit/87945f8b81c3976e0a984de59651ac39df535a1c/checks/80763048485/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
572           has_self: false,
573           has_late_bound_regions: None,

Note: some mismatched output was normalized before being compared
-               DefId(0:8 ~ dump_generics[c1f6]::NiceOfTheFoundation::U): 5,
-               DefId(0:7 ~ dump_generics[c1f6]::NiceOfTheFoundation::T): 4,
-               DefId(0:6 ~ dump_generics[c1f6]::NiceOfTheFoundation::N): 3,
-               DefId(0:5 ~ dump_generics[c1f6]::NiceOfTheFoundation::'_b): 2,
-               DefId(0:4 ~ dump_generics[c1f6]::NiceOfTheFoundation::'_a): 1,
-               DefId(0:3 ~ dump_generics[c1f6]::NiceOfTheFoundation): 0,
-               DefId(0:17 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::'a): 1,
-               DefId(0:16 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode): 0,
-               DefId(0:21 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::U): 5,
-               DefId(0:20 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::T): 4,
-               DefId(0:19 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::N): 3,
-               DefId(0:18 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::'b): 2,
-               DefId(0:34 ~ dump_generics[c1f6]::HaveTakenOver::N): 2,
-               DefId(0:33 ~ dump_generics[c1f6]::HaveTakenOver::'b): 1,
-               DefId(0:32 ~ dump_generics[c1f6]::HaveTakenOver::'a): 0,
-               DefId(0:36 ~ dump_generics[c1f6]::HaveTakenOver::U): 4,
-               DefId(0:35 ~ dump_generics[c1f6]::HaveTakenOver::T): 3,
-               DefId(0:59 ~ dump_generics[c1f6]::i_dare_you::N): 1,
-               DefId(0:57 ~ dump_generics[c1f6]::i_dare_you::'a): 0,
-               DefId(0:61 ~ dump_generics[c1f6]::i_dare_you::U): 3,
-               DefId(0:60 ~ dump_generics[c1f6]::i_dare_you::T): 2,
+               DefId(..): 5,
+               DefId(..): 3,
+               DefId(..): 0,
+               DefId(..): 1,
+               DefId(..): 5,
+               DefId(..): 2,
+               DefId(..): 1,
+               DefId(..): 4,
+               DefId(..): 3,
+               DefId(..): 2,
+               DefId(..): 0,
+               DefId(..): 4,
+               DefId(..): 2,
+               DefId(..): 4,
+               DefId(..): 1,
+               DefId(..): 3,
+               DefId(..): 1,
+               DefId(..): 3,
+               DefId(..): 0,


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args attributes/dump_generics.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/i686-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/attributes/dump_generics.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/i686-unknown-linux-gnu/stage2" "--target=i686-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/i686-unknown-linux-gnu/test/ui/attributes/dump_generics" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/i686-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: rustc_dump_generics: DefId(0:3 ~ dump_generics[c1f6]::NiceOfTheFoundation)
##[error]  --> /checkout/tests/ui/attributes/dump_generics.rs:8:1
   |
LL | trait NiceOfTheFoundation<'_a: '_a, '_b, const N: usize, T, U: Clone> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: Generics {
          parent: None,
          parent_count: 0,
          own_params: [
              GenericParamDef {
                  name: "Self",
                  def_id: DefId(0:3 ~ dump_generics[c1f6]::NiceOfTheFoundation),
                  index: 0,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
              GenericParamDef {
                  name: "'_a",
                  def_id: DefId(0:4 ~ dump_generics[c1f6]::NiceOfTheFoundation::'_a),
                  index: 1,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "'_b",
                  def_id: DefId(0:5 ~ dump_generics[c1f6]::NiceOfTheFoundation::'_b),
                  index: 2,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "N",
                  def_id: DefId(0:6 ~ dump_generics[c1f6]::NiceOfTheFoundation::N),
                  index: 3,
                  pure_wrt_drop: false,
                  kind: Const {
                      has_default: false,
                  },
              },
              GenericParamDef {
                  name: "T",
                  def_id: DefId(0:7 ~ dump_generics[c1f6]::NiceOfTheFoundation::T),
                  index: 4,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
              GenericParamDef {
                  name: "U",
                  def_id: DefId(0:8 ~ dump_generics[c1f6]::NiceOfTheFoundation::U),
                  index: 5,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
          ],
          param_def_id_to_index: {
              DefId(0:8 ~ dump_generics[c1f6]::NiceOfTheFoundation::U): 5,
              DefId(0:7 ~ dump_generics[c1f6]::NiceOfTheFoundation::T): 4,
              DefId(0:6 ~ dump_generics[c1f6]::NiceOfTheFoundation::N): 3,
              DefId(0:5 ~ dump_generics[c1f6]::NiceOfTheFoundation::'_b): 2,
              DefId(0:4 ~ dump_generics[c1f6]::NiceOfTheFoundation::'_a): 1,
              DefId(0:3 ~ dump_generics[c1f6]::NiceOfTheFoundation): 0,
          },
          has_self: true,
          has_late_bound_regions: None,
      }
  --> /checkout/tests/ui/attributes/dump_generics.rs:8:1
   |
LL | trait NiceOfTheFoundation<'_a: '_a, '_b, const N: usize, T, U: Clone> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: rustc_dump_generics: DefId(0:16 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode)
##[error]  --> /checkout/tests/ui/attributes/dump_generics.rs:28:1
   |
LL | pub trait IHopeItMadeLotsOfCode<'a, 'b, const N: usize, T, U: Clone>
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: Generics {
          parent: None,
          parent_count: 0,
          own_params: [
              GenericParamDef {
                  name: "Self",
                  def_id: DefId(0:16 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode),
                  index: 0,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
              GenericParamDef {
                  name: "'a",
                  def_id: DefId(0:17 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::'a),
                  index: 1,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "'b",
                  def_id: DefId(0:18 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::'b),
                  index: 2,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "N",
                  def_id: DefId(0:19 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::N),
                  index: 3,
                  pure_wrt_drop: false,
                  kind: Const {
                      has_default: false,
                  },
              },
              GenericParamDef {
                  name: "T",
                  def_id: DefId(0:20 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::T),
                  index: 4,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
              GenericParamDef {
                  name: "U",
                  def_id: DefId(0:21 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::U),
                  index: 5,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
          ],
          param_def_id_to_index: {
              DefId(0:17 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::'a): 1,
              DefId(0:16 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode): 0,
              DefId(0:21 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::U): 5,
              DefId(0:20 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::T): 4,
              DefId(0:19 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::N): 3,
              DefId(0:18 ~ dump_generics[c1f6]::IHopeItMadeLotsOfCode::'b): 2,
          },
          has_self: true,
          has_late_bound_regions: None,
      }
  --> /checkout/tests/ui/attributes/dump_generics.rs:28:1
   |
LL | pub trait IHopeItMadeLotsOfCode<'a, 'b, const N: usize, T, U: Clone>
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: rustc_dump_generics: DefId(0:22 ~ dump_generics[c1f6]::LookItsFromCorro)
##[error]  --> /checkout/tests/ui/attributes/dump_generics.rs:33:1
   |
LL | struct LookItsFromCorro<'a, 'b, const N: usize, T, U: Clone> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: Generics {
          parent: None,
          parent_count: 0,
          own_params: [
              GenericParamDef {
                  name: "'a",
                  def_id: DefId(0:23 ~ dump_generics[c1f6]::LookItsFromCorro::'a),
                  index: 0,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "'b",
                  def_id: DefId(0:24 ~ dump_generics[c1f6]::LookItsFromCorro::'b),
                  index: 1,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "N",
                  def_id: DefId(0:25 ~ dump_generics[c1f6]::LookItsFromCorro::N),
                  index: 2,
                  pure_wrt_drop: false,
                  kind: Const {
                      has_default: false,
                  },
              },
              GenericParamDef {
                  name: "T",
                  def_id: DefId(0:26 ~ dump_generics[c1f6]::LookItsFromCorro::T),
                  index: 3,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
              GenericParamDef {
                  name: "U",
                  def_id: DefId(0:27 ~ dump_generics[c1f6]::LookItsFromCorro::U),
                  index: 4,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
          ],
          param_def_id_to_index: {
              DefId(0:25 ~ dump_generics[c1f6]::LookItsFromCorro::N): 2,
              DefId(0:24 ~ dump_generics[c1f6]::LookItsFromCorro::'b): 1,
              DefId(0:23 ~ dump_generics[c1f6]::LookItsFromCorro::'a): 0,
              DefId(0:27 ~ dump_generics[c1f6]::LookItsFromCorro::U): 4,
              DefId(0:26 ~ dump_generics[c1f6]::LookItsFromCorro::T): 3,
          },
          has_self: false,
          has_late_bound_regions: None,
      }
  --> /checkout/tests/ui/attributes/dump_generics.rs:33:1
   |
LL | struct LookItsFromCorro<'a, 'b, const N: usize, T, U: Clone> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: rustc_dump_generics: DefId(0:31 ~ dump_generics[c1f6]::HaveTakenOver)
##[error]  --> /checkout/tests/ui/attributes/dump_generics.rs:40:1
   |
LL | enum HaveTakenOver<'a: 'a, 'b, const N: usize, T, U: Clone> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: Generics {
          parent: None,
          parent_count: 0,
          own_params: [
              GenericParamDef {
                  name: "'a",
                  def_id: DefId(0:32 ~ dump_generics[c1f6]::HaveTakenOver::'a),
                  index: 0,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "'b",
                  def_id: DefId(0:33 ~ dump_generics[c1f6]::HaveTakenOver::'b),
                  index: 1,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "N",
                  def_id: DefId(0:34 ~ dump_generics[c1f6]::HaveTakenOver::N),
                  index: 2,
                  pure_wrt_drop: false,
                  kind: Const {
                      has_default: false,
                  },
              },
              GenericParamDef {
                  name: "T",
                  def_id: DefId(0:35 ~ dump_generics[c1f6]::HaveTakenOver::T),
                  index: 3,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
              GenericParamDef {
                  name: "U",
                  def_id: DefId(0:36 ~ dump_generics[c1f6]::HaveTakenOver::U),
                  index: 4,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
          ],
          param_def_id_to_index: {
              DefId(0:34 ~ dump_generics[c1f6]::HaveTakenOver::N): 2,
              DefId(0:33 ~ dump_generics[c1f6]::HaveTakenOver::'b): 1,
              DefId(0:32 ~ dump_generics[c1f6]::HaveTakenOver::'a): 0,
              DefId(0:36 ~ dump_generics[c1f6]::HaveTakenOver::U): 4,
              DefId(0:35 ~ dump_generics[c1f6]::HaveTakenOver::T): 3,
          },
          has_self: false,
          has_late_bound_regions: None,
      }
  --> /checkout/tests/ui/attributes/dump_generics.rs:40:1
   |
LL | enum HaveTakenOver<'a: 'a, 'b, const N: usize, T, U: Clone> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: rustc_dump_generics: DefId(0:47 ~ dump_generics[c1f6]::TheFoundation)
##[error]  --> /checkout/tests/ui/attributes/dump_generics.rs:48:1
   |
LL | union TheFoundation<'a: 'a, 'b, const N: usize, T, U: Clone> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: Generics {
          parent: None,
          parent_count: 0,
          own_params: [
              GenericParamDef {
                  name: "'a",
                  def_id: DefId(0:48 ~ dump_generics[c1f6]::TheFoundation::'a),
                  index: 0,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "'b",
                  def_id: DefId(0:49 ~ dump_generics[c1f6]::TheFoundation::'b),
                  index: 1,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "N",
                  def_id: DefId(0:50 ~ dump_generics[c1f6]::TheFoundation::N),
                  index: 2,
                  pure_wrt_drop: false,
                  kind: Const {
                      has_default: false,
                  },
              },
              GenericParamDef {
                  name: "T",
                  def_id: DefId(0:51 ~ dump_generics[c1f6]::TheFoundation::T),
                  index: 3,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
              GenericParamDef {
                  name: "U",
                  def_id: DefId(0:52 ~ dump_generics[c1f6]::TheFoundation::U),
                  index: 4,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
          ],
          param_def_id_to_index: {
              DefId(0:51 ~ dump_generics[c1f6]::TheFoundation::T): 3,
              DefId(0:50 ~ dump_generics[c1f6]::TheFoundation::N): 2,
              DefId(0:49 ~ dump_generics[c1f6]::TheFoundation::'b): 1,
              DefId(0:48 ~ dump_generics[c1f6]::TheFoundation::'a): 0,
              DefId(0:52 ~ dump_generics[c1f6]::TheFoundation::U): 4,
          },
          has_self: false,
          has_late_bound_regions: None,
      }
  --> /checkout/tests/ui/attributes/dump_generics.rs:48:1
   |
LL | union TheFoundation<'a: 'a, 'b, const N: usize, T, U: Clone> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: rustc_dump_generics: DefId(0:56 ~ dump_generics[c1f6]::i_dare_you)
##[error]  --> /checkout/tests/ui/attributes/dump_generics.rs:55:1
   |
LL | const fn i_dare_you<'a: 'a, 'b, const N: usize, T, U>(you_can: &'a bool, _: &'b [(T, U); N]) {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: Generics {
          parent: None,
          parent_count: 0,
          own_params: [
              GenericParamDef {
                  name: "'a",
                  def_id: DefId(0:57 ~ dump_generics[c1f6]::i_dare_you::'a),
                  index: 0,
                  pure_wrt_drop: false,
                  kind: Lifetime,
              },
              GenericParamDef {
                  name: "N",
                  def_id: DefId(0:59 ~ dump_generics[c1f6]::i_dare_you::N),
                  index: 1,
                  pure_wrt_drop: false,
                  kind: Const {
                      has_default: false,
                  },
              },
              GenericParamDef {
                  name: "T",
                  def_id: DefId(0:60 ~ dump_generics[c1f6]::i_dare_you::T),
                  index: 2,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
              GenericParamDef {
                  name: "U",
                  def_id: DefId(0:61 ~ dump_generics[c1f6]::i_dare_you::U),
                  index: 3,
                  pure_wrt_drop: false,
                  kind: Type {
                      has_default: false,
                      synthetic: false,
                  },
              },
          ],
          param_def_id_to_index: {
              DefId(0:59 ~ dump_generics[c1f6]::i_dare_you::N): 1,
              DefId(0:57 ~ dump_generics[c1f6]::i_dare_you::'a): 0,
              DefId(0:61 ~ dump_generics[c1f6]::i_dare_you::U): 3,
              DefId(0:60 ~ dump_generics[c1f6]::i_dare_you::T): 2,
          },
          has_self: false,
          has_late_bound_regions: Some(
              /checkout/tests/ui/attributes/dump_generics.rs:55:29: 55:31 (#0),
          ),
      }
  --> /checkout/tests/ui/attributes/dump_generics.rs:55:1
   |
LL | const fn i_dare_you<'a: 'a, 'b, const N: usize, T, U>(you_can: &'a bool, _: &'b [(T, U); N]) {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: rustc_dump_generics: DefId(0:64 ~ dump_generics[c1f6]::IfYouNeed)
##[error]  --> /checkout/tests/ui/attributes/dump_generics.rs:66:1
   |
LL | trait IfYouNeed<'_a: '_a, '_b, const N: usize, T, U: Clone> {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: Generics {
          parent: None,
          parent_count: 0,
          own_params: [
              GenericParamDef {
                  name: "Self",
                  def_id: DefId(0:64 ~ dump_generics[c1f6]::IfYouNeed),
                  index: 0,
                  pure_wrt_drop: false,
```

</details>

--- a/compiler/rustc_middle/src/ty/generics.rs
+++ b/compiler/rustc_middle/src/ty/generics.rs
@@ -114,7 +114,7 @@ pub struct GenericParamCount {
 ///
 /// The ordering of parameters is the same as in [`ty::GenericArg`] (excluding child generics):
 /// `Self` (optionally), `Lifetime` params..., `Type` params...
-#[derive(Clone, Debug, TyEncodable, TyDecodable, StableHash)]
+#[derive(Clone, TyEncodable, TyDecodable, StableHash)]
 pub struct Generics {
     pub parent: Option<DefId>,
     pub parent_count: usize,
@@ -126,6 +126,23 @@ pub struct Generics {
 
     pub has_self: bool,
     pub has_late_bound_regions: Option<Span>,
+}
+
+impl std::fmt::Debug for Generics {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        // ironically, we get this warning because of what we're trying to fix.
+        #[expect(rustc::potential_query_instability)]
+        let mut stabilized_hashmap = self.param_def_id_to_index.iter().collect::<Vec<_>>();
+        stabilized_hashmap.sort_by_key(|(_, v)| **v);
+        f.debug_struct("Generics")
+            .field("parent", &self.parent)
+            .field("parent_count", &self.parent_count)
+            .field("own_params", &self.own_params)
+            .field("param_def_id_to_index", &stabilized_hashmap)
+            .field("has_self", &self.has_self)
+            .field("has_late_bound_regions", &self.has_late_bound_regions)
+            .finish()
+    }
 }
 
 impl<'tcx> rustc_type_ir::inherent::GenericsOf<TyCtxt<'tcx>> for &'tcx Generics {

--- a/tests/ui/attributes/dump_generics.stderr
+++ b/tests/ui/attributes/dump_generics.stderr
@@ -62,14 +62,32 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 4,
-              DefId(..): 0,
-              DefId(..): 2,
-              DefId(..): 3,
-              DefId(..): 1,
-              DefId(..): 5,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  0,
+              ),
+              (
+                  DefId(..),
+                  1,
+              ),
+              (
+                  DefId(..),
+                  2,
+              ),
+              (
+                  DefId(..),
+                  3,
+              ),
+              (
+                  DefId(..),
+                  4,
+              ),
+              (
+                  DefId(..),
+                  5,
+              ),
+          ],
           has_self: true,
           has_late_bound_regions: None,
       }
@@ -142,14 +160,32 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 2,
-              DefId(..): 0,
-              DefId(..): 4,
-              DefId(..): 5,
-              DefId(..): 3,
-              DefId(..): 1,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  0,
+              ),
+              (
+                  DefId(..),
+                  1,
+              ),
+              (
+                  DefId(..),
+                  2,
+              ),
+              (
+                  DefId(..),
+                  3,
+              ),
+              (
+                  DefId(..),
+                  4,
+              ),
+              (
+                  DefId(..),
+                  5,
+              ),
+          ],
           has_self: true,
           has_late_bound_regions: None,
       }
@@ -212,13 +248,28 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 3,
-              DefId(..): 4,
-              DefId(..): 2,
-              DefId(..): 0,
-              DefId(..): 1,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  0,
+              ),
+              (
+                  DefId(..),
+                  1,
+              ),
+              (
+                  DefId(..),
+                  2,
+              ),
+              (
+                  DefId(..),
+                  3,
+              ),
+              (
+                  DefId(..),
+                  4,
+              ),
+          ],
           has_self: false,
           has_late_bound_regions: None,
       }
@@ -281,13 +332,28 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 1,
-              DefId(..): 3,
-              DefId(..): 4,
-              DefId(..): 2,
-              DefId(..): 0,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  0,
+              ),
+              (
+                  DefId(..),
+                  1,
+              ),
+              (
+                  DefId(..),
+                  2,
+              ),
+              (
+                  DefId(..),
+                  3,
+              ),
+              (
+                  DefId(..),
+                  4,
+              ),
+          ],
           has_self: false,
           has_late_bound_regions: None,
       }
@@ -350,13 +416,28 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 4,
-              DefId(..): 3,
-              DefId(..): 1,
-              DefId(..): 2,
-              DefId(..): 0,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  0,
+              ),
+              (
+                  DefId(..),
+                  1,
+              ),
+              (
+                  DefId(..),
+                  2,
+              ),
+              (
+                  DefId(..),
+                  3,
+              ),
+              (
+                  DefId(..),
+                  4,
+              ),
+          ],
           has_self: false,
           has_late_bound_regions: None,
       }
@@ -412,12 +493,24 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 0,
-              DefId(..): 3,
-              DefId(..): 2,
-              DefId(..): 1,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  0,
+              ),
+              (
+                  DefId(..),
+                  1,
+              ),
+              (
+                  DefId(..),
+                  2,
+              ),
+              (
+                  DefId(..),
+                  3,
+              ),
+          ],
           has_self: false,
           has_late_bound_regions: Some(
               $DIR/dump_generics.rs:55:29: 55:31 (#0),
@@ -492,14 +585,32 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 4,
-              DefId(..): 2,
-              DefId(..): 0,
-              DefId(..): 5,
-              DefId(..): 1,
-              DefId(..): 3,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  0,
+              ),
+              (
+                  DefId(..),
+                  1,
+              ),
+              (
+                  DefId(..),
+                  2,
+              ),
+              (
+                  DefId(..),
+                  3,
+              ),
+              (
+                  DefId(..),
+                  4,
+              ),
+              (
+                  DefId(..),
+                  5,
+              ),
+          ],
           has_self: true,
           has_late_bound_regions: None,
       }
@@ -562,13 +673,28 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 4,
-              DefId(..): 0,
-              DefId(..): 2,
-              DefId(..): 3,
-              DefId(..): 1,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  0,
+              ),
+              (
+                  DefId(..),
+                  1,
+              ),
+              (
+                  DefId(..),
+                  2,
+              ),
+              (
+                  DefId(..),
+                  3,
+              ),
+              (
+                  DefId(..),
+                  4,
+              ),
+          ],
           has_self: false,
           has_late_bound_regions: None,
       }
@@ -587,7 +713,7 @@ note: Generics {
           parent: None,
           parent_count: 0,
           own_params: [],
-          param_def_id_to_index: {},
+          param_def_id_to_index: [],
           has_self: false,
           has_late_bound_regions: None,
       }
@@ -619,9 +745,12 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 6,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  6,
+              ),
+          ],
           has_self: true,
           has_late_bound_regions: None,
       }
@@ -642,7 +771,7 @@ note: Generics {
           ),
           parent_count: 6,
           own_params: [],
-          param_def_id_to_index: {},
+          param_def_id_to_index: [],
           has_self: true,
           has_late_bound_regions: None,
       }
@@ -677,9 +806,12 @@ note: Generics {
                   },
               },
           ],
-          param_def_id_to_index: {
-              DefId(..): 6,
-          },
+          param_def_id_to_index: [
+              (
+                  DefId(..),
+                  6,
+              ),
+          ],
           has_self: true,
           has_late_bound_regions: Some(
               $DIR/dump_generics.rs:21:9: 21:10 (#0),

